rust/phoenix-channel/src/login_url.rs

```diff
@@ -44,6 +47,7 @@ impl LoginUrl {
         device_id: String,
         device_name: Option<String>,
         public_key: [u8; 32],
+        additional_query_params: &HashMap<String, String>,
```

Comment:
1. Do we really want to pass around string => string maps? Why not make this typed, like a `struct`?
2. I'd like to discuss whether this information shouldn't go into the `User-Agent` instead (https://github.com/firezone/firezone/issues/6837#issuecomment-2384216972). That is where we have it currently.

Reply:
1. I was thinking that string => string maps would give us the most flexibility as this information will come from different places such as clients with different fields
2. that seems to be overloading the user-agent, the device id is the closest to this which is called `external_id` in the query params. Also it's easier to parse and add fields there rather than the user-agent which doesn't have a fixed structure.

Reply:
> I was thinking that string => string maps would give us the most flexibility as this information will come from different places such as clients with different fields

I mean, yeah. With that argument we don't need to be using any types anywhere :)

At the very least, we should make an enum for all the keys but I'd prefer a struct that models out this information, perhaps all query params that we can set?

Reply:
Using enums sounds good. But if we use a `DeviceInfo` struct it will have all `Option` fields which we would have to check one by one to add to the query params.

Reply:
> But if we use a `DeviceInfo` struct it will have all `Option` fields which we would have to check one by one to add to the query params.

What be so bad? At the moment we have a lot of `Option`s as parameters :man_shrugging: